### PR TITLE
Support column-based insert data in message stream

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
issue: #15794

### CODE REVIEW

File: internal/datanode/flow_graph_insert_buffer_node.go [hunk 1]
```diff
@@ -677,9 +633,262 @@ func (ibNode *insertBufferNode) bufferInsertMsg(msg *msgstream.InsertMsg, endPos
...
+	}
+
+	return primaryField, nil
+}
+
+// insertMsgToInsertData parses the insert msg and then fill the idata.
+func insertMsgToInsertData(msg *msgstream.InsertMsg,
+	collSchema *schemapb.CollectionSchema,
+	idata *storage.InsertData) (primaryField *storage.Int64FieldData, err error) {
+	if msg.RowBasedFormat() {
```
Comment (xige-16) [inline internal/datanode/flow_graph_insert_buffer_node.go:825]: add log to print data format?

File: internal/msgstream/msg.go [hunk 2]
```diff
@@ -171,6 +174,90 @@ func (it *InsertMsg) Unmarshal(input MarshalType) (TsMsg, error) {
 	return insertMsg, nil
 }
 
+func (it *InsertMsg) RowBasedFormat() bool {
```
Comment (czs007) [inline internal/msgstream/msg.go:177]: IsRowBased()

File: internal/msgstream/msg.go [hunk 3]
```diff
@@ -171,6 +174,90 @@ func (it *InsertMsg) Unmarshal(input MarshalType) (TsMsg, error) {
...
+	return it.GetVersion() == internalpb.InsertRequestFormat_RowBased
+}
+
+func (it *InsertMsg) ColumnBasedFormat() bool {
```
Comment (czs007) [inline internal/msgstream/msg.go:181]: IsColumnBased()

File: internal/proto/internal.proto [hunk 4]
```diff
@@ -128,6 +128,12 @@ message CreateIndexRequest {
   repeated common.KeyValuePair extra_params = 8;
 }
 
+enum InsertRequestFormat {
```
Comment (czs007) [inline internal/proto/internal.proto:131]: Better to rename to InsertRequestVersion

File: internal/proxy/task.go [hunk 5]
```diff
@@ -97,6 +95,9 @@ const (
 	minFloat32 = -1 * float32(math.MaxFloat32)
 )
 
+// TODO: better to be configurable.
+var columnDataFormatInStream = true
```
Comment (czs007) [inline internal/proxy/task.go:99]: no need to be configurable.

There is no reason to still write row-wise data.

File: internal/proxy/task.go [hunk 6]
```diff
@@ -763,13 +530,23 @@ func (it *insertTask) PreExecute(ctx context.Context) error {
 	}
 
 	tr := timerecord.NewTimeRecorder("ColumnToRow")
-	err = it.transferColumnBasedRequestToRowBasedData()
-	if err != nil {
-		return err
+	if !columnDataFormatInStream {
```
Comment (czs007) [inline internal/proxy/task.go:533]: On the proxy side, delete the logic for maintaining the row-based data format.

File: internal/proxy/task.go [hunk 7]
```diff
@@ -954,6 +738,11 @@ func (it *insertTask) _assignSegmentID(stream msgstream.MsgStream, pack *msgstre
 					SegmentID:      segmentID,
 					ShardName:      channelNames[key],
 				}
+				if columnDataFormatInStream {
```
Comment (czs007) [inline internal/proxy/task.go:741]: remove this check.

File: internal/msgstream/msg.go [hunk 8]
```diff
@@ -171,6 +174,90 @@ func (it *InsertMsg) Unmarshal(input MarshalType) (TsMsg, error) {
...
+}
+
+func (it *InsertMsg) NRows() uint64 {
+	if it.RowBasedFormat() {
+		return uint64(len(it.RowData))
+	}
+	return it.InsertRequest.GetNumRows()
+}
+
+func (it *InsertMsg) CheckAligned() bool {
```
Comment (xiaofan-luan) [inline internal/msgstream/msg.go:192]: Bingo! very smart function

**Comment by DragonDriver** [comment]
rerun ut

**Comment by DragonDriver** [comment]
rerun ut

**Comment by xiaofan-luan** [comment]
much clear and unified, thanks @DragonDriver

**Pull request merged**